plugins/modules/system/sysrc.py

```diff
@@ -0,0 +1,265 @@
+                return
+
+        self.changed = True
+
+    def value_present(self):
+        if self.contains():
+            return
+
+        if self.module.check_mode:
+            self.changed = True
+            return
+
+        setstring = '%s+=%s%s' % (self.name, self.delim, self.value)
+        (rc, out, err) = self.run_sysrc(setstring)
+        if out.find("%s:" % self.name) == 0:
+            values = out.split(' -> ')[1].strip().split(self.delim)
+            if self.value in values:
+                self.changed = True
+
+    def value_absent(self):
+        if not self.contains():
+            return
+
+        if self.module.check_mode:
+            self.changed = True
+            return
+
+        setstring = '%s-=%s%s' % (self.name, self.delim, self.value)
+        (rc, out, err) = self.run_sysrc(setstring)
+        if out.find("%s:" % self.name) == 0:
+            values = out.split(' -> ')[1].strip().split(self.delim)
+            if self.value not in values:
+                self.changed = True
+
+    def run_sysrc(self, *args):
+        cmd = [self.sysrc, '-f', self.path]
+        if self.jail:
+            cmd += ['-j', self.jail]
+        cmd.extend(args)
+
+        (rc, out, err) = self.module.run_command(cmd)
+        if self.module._verbosity > 0:
+            self.cmd = cmd
+
+        if self.module._verbosity >= 4:
+            self.cmd_output = out
+
+        return (rc, out, err)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type='str', required=True),
+            value=dict(type='str', default=None),
+            state=dict(type='str', default='present', choices=['absent', 'present', 'value_present', 'value_absent']),
+            path=dict(type='str', default='/etc/rc.conf'),
+            delim=dict(type='str', default=' '),
+            jail=dict(type='str', default=None),
+        ),
+        supports_check_mode=True,
+    )
+
+    name = module.params.pop('name')
+    # OID style names are not supported
+    if not re.match('^[a-zA-Z0-9_]+$', name):
+        module.fail_json(
+            msg="Name may only contain alpha-numeric and underscore characters"
+        )
+
+    value = module.params.pop('value')
+    state = module.params.pop('state')
+    path = module.params.pop('path')
+    delim = module.params.pop('delim')
+    jail = module.params.pop('jail')
+    result = dict(
+        name=name,
+        state=state,
+        value=value,
+        path=path,
+        delim=delim,
+        jail=jail
+    )
+
+    rc_value = Sysrc(module, name, value, path, delim, jail)
+
+    if module._verbosity >= 4:
+        result['existed'] = rc_value.exists()
+
+    if state == 'present':
+        rc_value.present()
+    elif state == 'absent':
+        rc_value.absent()
+    elif state == 'value_present':
+        rc_value.value_present()
+    elif state == 'value_absent':
+        rc_value.value_absent()
+
+    if module._verbosity > 0:
```

Comment:
Hi @dlundgren I didn't investigate this thoroughly, but it _seems_ that this value doesn't ever get negative. In any case, apparently the semantics of this statement is to _always_ add the command to the output, would that be a fair assessment? If yes, then why test verbosity anyway?

BTW, I second the other comment: this is how a new module PR should look like. :-) Thanks for putting it together.

Reply:
You're right, I ran a couple of tests, and I don't think it matters if we remove this verbosity check.

Would `1` be the  recommended verbosity for the generated command, or should it be higher?

For example, at verbosity `4` I return the command output, if any.

Reply:
IMHO modules' output (as in the dict of output values, as we have in this case) shouldn't  care all that much about verbosity, given that those values are supposed to be processed (or not) by other automared tasks. So, personally I would dump everything every time on the exit. (That sounded better in my mind)

And I am not sure there is a recommendation on that, so YMMV.

Reply:
In general it is better to always output everything. Then users can decide on what they want to do with it, and they won't be surprised when they saw something exists when running with higher verbosity setting, use it in their role, and suddenly find out it isn't there when they run with default verbosity.

Reply:
I'm inclined to recommend that the verbosity level outputs be removed, instead of keeping them. 

From a users perspective, I don't think I would use them unless I'm debugging an issue with the module.

If the outputs are kept, should they be documented in the `RETURN` block?

Reply:
Let's remove the verbosity check then. Documenting them would be nice, but if they are for debug reasons I *think* it is OK to omit them.

Reply:
I _think_ the _them_ to be omitted were the verbosity checks, not the outputs 😮 ... @felixfontein should be online in another couple of hours or so to confirm. Personally, I'd rather have the outputs, unconditionally.

Reply:
Yes, I was suggesting to keep the output unconditionally. I'm also fine with them being gone, as long as `module._verbosity` is not used :)